x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -517,6 +620,41 @@ And user should NOT see the Rule Updates table
 
 TODO: add scenarios
 
+#### **Scenario: User can preview a rule before upgrading**
+
+**Automation**: 1 e2e test
+
+```Gherkin
+Given at least 2 prebuilt rules are installed in Kibana
+And for at least 2 of the installed rules there are new versions available
+And user is on the Rule Management page
+When user opens the Rule Updates table
+Then all rules available for upgrade should be displayed in the table
+And user should be able to open a preview for a rule
+When the preview is open the user should be able to close it
+Then the user should be able to open a preview for another rule
+When the preview is open the user should be able to upgrade a rule using an action in the rule preview
+Then success message should be displayed after upgrade
+And the upgraded rule should be removed from the table
+And user should see the number of rules available to upgrade as 1
+```
```

Comment:
This is probably nitpicking.

```Gherkin
When the preview is open the user should be able to close it
```

```Gherkin
When the preview is open the user should be able to upgrade a rule using an action in the rule preview
```

I think this wording is a bit non-canonical. I think `when` should not be used as `if`, as in the examples above. In BDD, `when` means action (verb) and `then` means consequence and assertion. Actions can be actively _performed by a subject_ (a user, an app, a process, a workflow) or can just _happen_ (events). For example:

- When user (subject) opens the Rule Updates table (action)
- When I (subject) click this button (action)
- When XYZ happens (event)
- Then the BlahBlah callout should be displayed
- Then the rule should be installed and show up in the table

The reworded scenario can look like this:

```Gherkin
Given 2 prebuilt rules are installed in Kibana
And for these 2 installed rules there are new versions available
And user is on the Rule Management page
When user opens the Rule Updates table
Then all rules available for upgrade should be displayed in the table
When user opens the rule preview for the 1st rule
Then the preview should open
When user closes the preview
Then it should disappear
When user opens the rule preview for the 2nd rule
Then the preview should open
When user upgrades the rule using a CTA in the rule preview
Then the 2nd rule should be upgraded to the latest version
And a success message should be displayed after upgrade
And the 2nd rule should be removed from the Rule Updates table
And user should see the number of rules available to upgrade as 1
```

This canonical wording is:

- more precise; it tells how exactly the app should behave
- much easier to automate and map to real code blocks; because in imperatively written tests we don't have declarative "if-then rules", we have actions and assertions (as in `Arrange, Act, Assert`)

So I went ahead and adjusted the scenarios accordingly. All that said:

- it is all arguable: there are pros for having declarative "if-then rules", because declarative specs are easier to read and understand
- the original scenarios were shorter and didn't have any issues with readability

LMK if you have any concerns with the edits I pushed to the PR, and we will reword this.